test/routing-tables.test.js

```diff
@@ -120,6 +120,24 @@ describe('RoutingTables', function () {
         undefined)
     })
 
+    it('creates a route with a custom prefix, if supplied', function () {
+      const route = {
+        target_prefix: 'prefix.',
+        source_ledger: ledgerB,
+        destination_ledger: ledgerC,
+        source_account: ledgerC + 'mary',
+        min_message_window: 1,
+        points: [ [0, 0], [50, 60] ]
+      }
+
+      // will be stored in destinations under 'prefix.'
+      assert.equal(this.tables.addRoute(route), true)
+
+      assertSubset(
+        this.tables.sources.get(ledgerA).destinations.get('prefix.').get(ledgerC + 'mary'),
+        { destinationLedger: ledgerC })
```

Comment:
In this case, the `targetPrefix` is not preserved in the final route due to joining the routes (although it still is used for the entry in the destinations table). Should `Route.join` preserve custom route prefixes? If you joined two routes with custom `targetPrefix`es, how do you decide which to preserve in the final route?